bson/bsoncodec/default_value_encoders.go

```diff
@@ -289,29 +289,44 @@ func (dve DefaultValueEncoders) MapEncodeValue(ec EncodeContext, vw bsonrw.Value
 // struct codec.
 func (dve DefaultValueEncoders) mapEncodeValue(ec EncodeContext, dw bsonrw.DocumentWriter, val reflect.Value, collisionFn func(string) bool) error {
 
-	encoder, err := ec.LookupEncoder(val.Type().Elem())
-	if err != nil {
-		return err
+	encoder, encErr := ec.LookupEncoder(val.Type().Elem())
```

Comment:
nit: I'm not sure what the perf of these reflection calls is, but maybe it'd be better to call `val.Type().Elem()` only once and store the result?

Reply:
done